src/replication/adapter/helmhub/adapter.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helmhub
+
+import (
+	"errors"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	adp "github.com/goharbor/harbor/src/replication/adapter"
+	"github.com/goharbor/harbor/src/replication/model"
+)
+
+func init() {
+	if err := adp.RegisterFactory(model.RegistryTypeHelmHub, func(registry *model.Registry) (adp.Adapter, error) {
+		return newAdapter(registry)
+	}); err != nil {
+		log.Errorf("failed to register factory for %s: %v", model.RegistryTypeHelmHub, err)
+		return
+	}
+	log.Infof("the factory for adapter %s registered", model.RegistryTypeHelmHub)
+}
+
+type adapter struct {
+	registry *model.Registry
+	client   *Client
+}
+
+func newAdapter(registry *model.Registry) (*adapter, error) {
+	return &adapter{
+		registry: registry,
+		client:   NewClient(registry),
+	}, nil
+}
+
+func (a *adapter) Info() (*model.RegistryInfo, error) {
```

Comment:
From the implementation code, we can see that the helm hub adapter does not support push mode. Have we considered declaring such capability too?

@mmpei @ywk253100

Reply:
it seams that there are some restrictions in UI. maybe hard code now.
agree with @ywk253100